src/main/java/seedu/address/model/person/Person.java

```diff
@@ -61,16 +61,16 @@ public Address getAddress() {
     }
 
     /**
-     * Returns true if both persons have the same name.
+     * Returns true if both clients have the same address.
      * This defines a weaker notion of equality between two persons.
      */
-    public boolean isSamePerson(Person otherPerson) {
+    public boolean isSameClient(Person otherPerson) {
         if (otherPerson == this) {
             return true;
         }
 
         return otherPerson != null
-                && otherPerson.getName().equals(getName());
+                && otherPerson.getAddress().equals(getAddress());
```

Comment:
I see a potential bug here when simply comparing the two addresses.
Assuming we have one Client: John Doe; Address: 311, Clementi Ave 2
We can have a duplicate: John Doe; Address: 311, Clementi Avenue 2.

This can apply for other comparisons depending on how the address is formatted.

As discussed, some potential methods could be using the User Guide as a disclaimer instead, as there are too many permutations that can happen.